test/docker-osx-dev.bats

```diff
@@ -417,6 +417,12 @@ export NEW_ENV_VARIABLE_2=VALUE2"
   assert_output '/some/path'
 }
 
+@test "find_path_to_sync_parent should match paths with weird characters" {
+  export PATHS_TO_SYNC='/some/path2()$HI'
```

Comment:
OK, so the `eval` is there so that paths with `~` get interpreted correctly, as per 438019f3fb543c7499c5057508bb23e8206b50aa. Using single quotes will hopefully not have any impact on this, so this seems like a reasonable fix. 

The issue in #102 does seem oddly similar, but there is no eval directly around it. Probably best to look into & fix that one separately, so I'll merge this one for now. Thanks!